lib/bson/float.rb

```diff
@@ -57,37 +57,26 @@ def to_bson(buffer = ByteBuffer.new, validating_keys = Config.validating_keys?)
     #
     # @return [ Hash | Float ] The extended json representation.
     def as_extended_json(**options)
-      case infinite?
-      when 1
-        {'$numberDouble' => 'Infinity'}
-      when -1
-        {'$numberDouble' => '-Infinity'}
-      else
-        if nan?
-          {'$numberDouble' => 'NaN'}
-        else
-          if options[:mode] == :relaxed || options[:mode] == :legacy
-            self
-          else
-            value = if BSON::Environment.jruby?
-              # Hack to make bson corpus spec tests pass.
-              # JRuby serializes -1.2345678901234568e+18 as
-              # -1234567890123456770.0, which is valid but differs from MRI
-              # serialization. Extended JSON spec does not define precise
-              # stringification of floats.
-              # https://jira.mongodb.org/browse/SPEC-1536
-              if abs > 1e15
-                '%.17g' % to_s
+      value = if infinite? == 1
+                'Infinity'
+              elsif infinite? == -1
+                '-Infinity'
+              elsif nan?
+                'NaN'
+              elsif options[:mode] == :relaxed || options[:mode] == :legacy
+                self
```

Comment:
@johnnyshields I believe all tests are failing because this line wraps numeric literals into `$numberDouble` which shouldn't be happening.

Reply:
Fixed, thanks!